src/scipp/html/formatting_datagroup_html.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+# Original source from
+# https://github.com/jsignell/xarray/blob/1d960933ab252e0f79f7e050e6c9261d55568057/xarray/core/formatting_html.py
+
+import uuid
+from html import escape
+from string import Template
+from typing import Union
+
+import numpy as np
+
+from ..core.cpp_classes import DataArray, Dataset, Variable
+from ..core.data_group import DataGroup
+from ..units import dimensionless
+from .resources import load_atomic_tpl, load_collapsible_tpl, load_dg_detail_tpl, \
+    load_dg_repr_tpl, load_dg_style
+
+
+def _make_shape_repr(var: Union[Variable, DataArray, Dataset, DataGroup]):
+    """Returns HTML Component that represents the shape of ``var``"""
+    shape_list = "".join(f"<li>{escape(str(dim))}: {size}</li>"
+                         for dim, size in var.sizes.items())
+    return f"<ul class='sc-dim-list'>{shape_list}</ul>"
+
+
+def _make_dataset_summary(ds: Dataset):
+    """Returns (partial) information of Dataset object items"""
+    key_type_str = [
+        ": ".join([name, type(value).__name__]) for name, value in ds.items()
+    ]
+    if len(key_type_str) > 3:
+        key_type_str = key_type_str[:2] + ["..."] + key_type_str[-1:]
+    return "Dataset{" + ", ".join(key_type_str) + "}"
+
+
+def _summarize_atomic_variable(var, name: str, depth: int = 0):
+    """Returns HTML Component that contains summary of ``var``"""
+    shape_repr = ''
+    unit = ''
+    dtype_str = ''
+    preview = ''
+    parent_obj_str = ''
+    objtype_str = type(var).__name__
+    if isinstance(var, (Dataset, DataArray, Variable, DataGroup)):
+        parent_obj_str = "scipp"
+        shape_repr = _make_shape_repr(var)
+        if isinstance(var, Dataset):
+            preview = _make_dataset_summary(var)
+        else:
+            if var.unit is not None:
+                unit = '𝟙' if var.unit == dimensionless else str(var.unit)
+            dtype_str = str(var.dtype)
+            preview = str(var.value) if len(var.dims) == 0 else ""
+    else:
+        if isinstance(var, np.ndarray):
+            shape_repr = escape("()")
```

Comment:
Shouldn't we use `()` also for all other non-scipp items, instead of an empty string? Or consistently use and empty string, including the Numpy case?

Reply:
I would rather use `()` for both. It was a bit hard to read when only some of them are empty. I will update it